static/app/views/dashboardsV2/gridLayout/dashboard.tsx

```diff
@@ -0,0 +1,95 @@
+import 'react-grid-layout/css/styles.css';
+import 'react-resizable/css/styles.css';
+
+import React from 'react';
+import RGL, {WidthProvider} from 'react-grid-layout';
+
+import withApi from 'app/utils/withApi';
+import withGlobalSelection from 'app/utils/withGlobalSelection';
+
+import AddWidget, {ADD_WIDGET_BUTTON_DRAG_ID} from '../addWidget';
+import {Dashboard as DnDKitDashboard} from '../dashboard';
+import {DisplayType, MAX_WIDGETS, Widget} from '../types';
+
+import SortableWidget from './sortableWidget';
+import {DRAG_HANDLE_ID} from './widgetCard';
+
+const NUM_COLS = 6;
+const ROW_HEIGHT = 120;
+const WIDGET_MARGINS = [16, 16];
+const ADD_BUTTON_POSITION = {
+  x: 0,
+  y: Number.MAX_VALUE,
+  w: 2,
+  h: 1,
+  isResizable: false,
+};
+const DEFAULT_WIDGET_WIDTH = 2;
+
+const GridLayout = WidthProvider(RGL);
+
+class Dashboard extends DnDKitDashboard {
```

Comment:
I would typically recommend against inheritance, and instead prefer composition. Can we do that here?

Reply:
Summarizing our conversation, we decided to avoid the inheritance and chose to copy the files over and modify the parts necessary for supporting the new layout. Instead, copying the files over is simple and avoids any difficulties down the road for promoting the new dashboard